src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -235,9 +248,18 @@ def _get_node_maps(self, cores, gpus, threads_per_proc):
         core_map = list()
         gpu_map  = list()
 
+        # Question: The following two lines are checking if the number of
+        # available cores is a multiple of the threads per CPU process and then 
+        # finds the number of CPU processes. Why not input the number of CPU
+        # processes to this function?
         assert(not len(cores) % threads_per_proc)
         n_procs =  len(cores) / threads_per_proc
 
+
+        # core_map is a list of list of core ids.
+        # core_map = [cores_for_proc_1,cores_for_proc_2,....., cores_for_proc_N]
+        # cores_for_proc_x = [core_for_thread_1, core_for_thread_2,.., 
+        #                       core_for_thread_M]
         idx = 0
```

Comment:
This is documented here: https://github.com/radical-cybertools/radical.pilot/blob/feature/gpu/src/radical/pilot/agent/scheduler/base.py#L113
Instead of having two places with that documentation, would you mind adding a reference / link?

Reply:
Makes sense. Done! Added some more detail in the base class.